### PR TITLE
VL-25_Date-&-color-pickers-new-composables_Vitalii-Dudnik

### PR DESCRIPTION
**Pull request by KinduD21**
https://ilevel.atlassian.net/jira/software/c/projects/VL/boards/11?selectedIssue=VL-25

### CODE REVIEW

File: src/ui.form-calendar/configs/default.config.js [hunk 1]
```diff
@@ -32,11 +32,11 @@ export default /*tw*/ {
   currentDay: "hover:bg-brand-100 border-2 border-brand-900",
   anotherMonthDay: "hover:bg-brand-100 text-brand-400",
   monthView: "grid grid-cols-4 items-center justify-center",
-  month: "mx-auto flex h-12 w-full rounded-lg hover:!bg-brand-900 hover:text-white",
+  month: "{UButton} mx-auto flex h-12 w-full rounded-lg hover:!bg-brand-900 hover:text-white",
   selectedMonth: "bg-brand-900 hover:bg-brand-900 text-white",
   activeMonth: "bg-brand-100",
   yearView: "grid grid-cols-4 items-center justify-center",
-  year: "mx-auto flex h-12 w-full rounded-lg hover:!bg-brand-900 hover:text-white",
+  year: "{UButton} mx-auto flex h-12 w-full rounded-lg hover:!bg-brand-900 hover:text-white",
```
Comment (itsJohnnyGrid) [inline src/ui.form-calendar/configs/default.config.js:39]: Calendar component have much more nested Vueless components. Please check `.vue` files and fix it.

File: src/ui.form-calendar/index.vue [hunk 2]
```diff
@@ -287,12 +287,44 @@ const props = defineProps({
...
   "update:modelValue",
+
+  /**
+   * Triggers when calendar view changes.
+   * @property {string} value
+   */
   "update:view",
+
+  /**
+   * Triggers when calendar view changes.
```
Comment (itsJohnnyGrid) [inline src/ui.form-calendar/index.vue:303]: Triggers when date value changes.

File: src/ui.form-calendar/index.vue [hunk 3]
```diff
@@ -287,12 +287,44 @@ const props = defineProps({
...
 const emit = defineEmits([
+  /**
+   * Triggers when date value changes.
+   * @property {object} newDate
+   */
   "update:modelValue",
+
+  /**
+   * Triggers when calendar view changes.
+   * @property {string} value
```
Comment (itsJohnnyGrid) [inline src/ui.form-calendar/index.vue:298]: are you sure?? maby `view`?

File: src/ui.form-calendar/index.vue [hunk 4]
```diff
@@ -287,12 +287,44 @@ const props = defineProps({
...
+  /**
+   * Triggers when calendar view changes.
+   * @property {object} value
+   */
   "input",
+
+  /**
+   * Triggers when calendar date is submitted.
+   */
   "submit",
```
Comment (itsJohnnyGrid) [inline src/ui.form-calendar/index.vue:311]: How it works?

File: src/ui.form-calendar/index.vue [hunk 5]
```diff
@@ -287,12 +287,44 @@ const props = defineProps({
...
+
+  /**
+   * Triggers when arrow keys are used to change calendar date.
+   */
   "keydown",
+
+  /**
+   * Triggers when user date changes.
```
Comment (itsJohnnyGrid) [inline src/ui.form-calendar/index.vue:319]: not clear description.

File: src/ui.form-calendar/index.vue [hunk 6]
```diff
@@ -287,12 +287,44 @@ const props = defineProps({
...
+   * @property {string} value
+   */
   "userDateChange",
+
+  /**
+   * Triggers when formatted date changes.
```
Comment (itsJohnnyGrid) [inline src/ui.form-calendar/index.vue:325]: not clear description.

File: src/ui.form-color-picker/configs/default.config.js [hunk 7]
```diff
@@ -14,12 +14,13 @@ export default /*tw*/ {
   },
   radio: {
     radio: {
-      base: "cursor-pointer bg-{color}-500 border-{color}-500 hover:border-{color}-500",
+      base: "{URadio} cursor-pointer bg-{color}-500 border-{color}-500 hover:border-{color}-500",
```
Comment (itsJohnnyGrid) [inline src/ui.form-color-picker/configs/default.config.js:17]: base: nesting no needed here

File: src/ui.form-date-picker/index.vue [hunk 8]
```diff
@@ -215,7 +218,19 @@ const props = defineProps({
   },
 });
 
-const emit = defineEmits(["update:modelValue", "input"]);
+const emit = defineEmits([
+  /**
+   * Triggers when date input value changes.
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker/index.vue:223]: Triggers when date picker value changes.

File: src/ui.form-date-picker/index.vue [hunk 9]
```diff
@@ -215,7 +218,19 @@ const props = defineProps({
...
+   * @property {string} value
+   */
+  "update:modelValue",
+
+  /**
+   * Triggers when date input value changes.
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker/index.vue:229]: Triggers when date picker value changes.

File: src/ui.form-date-picker/composables/attrs.composable.js [hunk 10]
```diff
@@ -1,80 +1,98 @@
...
-  const cvaCalendarWrapper = cva({
-    base: calendar.wrapper.base,
-    variants: calendar.wrapper.variants,
-    compoundVariants: calendar.wrapper.compoundVariants,
-  });
-
   const calendarWrapperClasses = computed(() =>
-    cvaCalendarWrapper({
+    cva(config.value.calendar.wrapper)({
       openDirectionY: openDirectionY.value,
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker/composables/attrs.composable.js:17]: computed inside computed no needed, add ternary directly here

File: src/ui.form-date-picker/composables/attrs.composable.js [hunk 11]
```diff
@@ -1,80 +1,98 @@
...
   const openDirectionY = computed(() => (isTop.value ? POSITION.top : POSITION.bottom));
   const openDirectionX = computed(() => (isRight.value ? POSITION.right : POSITION.left));
 
-  const cvaCalendarWrapper = cva({
-    base: calendar.wrapper.base,
-    variants: calendar.wrapper.variants,
-    compoundVariants: calendar.wrapper.compoundVariants,
-  });
-
   const calendarWrapperClasses = computed(() =>
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker/composables/attrs.composable.js:15]: computed no needed at all. you can do the same inside if statement in the loop

File: src/ui.form-date-picker/composables/attrs.composable.js [hunk 12]
```diff
@@ -1,80 +1,98 @@
...
+
+      attrs[`${key}Attrs`] = computed(() => ({
+        ...calendarAttrs.value,
+        class: cx([calendarWrapperClasses.value, calendarAttrs.value.class]),
+      }));
+
+      // This watcher rewrites default calendar locales with datepicker range locales
+      // Watcher will not rewrite custom calendar locales
+      watchEffect(() => {
+        if (!calendarAttrs.value.config) {
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker/composables/attrs.composable.js:59]: here and in all places below you need to modify not cashed variable `calendarAttrs` but current variable **attrs[`${key}Attrs`]**

File: src/ui.form-date-picker-range/index.vue [hunk 13]
```diff
@@ -423,7 +423,13 @@ const props = defineProps({
 
 const inputRangeFormat = "d.m.Y";
 
-const emit = defineEmits(["update:modelValue"]);
+const emit = defineEmits([
+  /**
+   * Triggers when date input value changes.
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/index.vue:428]: Triggers when date picker range value changes.

File: src/ui.form-date-picker-range/configs/default.config.js [hunk 14]
```diff
@@ -61,7 +61,7 @@ export default /*tw*/ {
       description: "hidden",
     },
     input: {
-      base: "text-sm border-brand-100 px-4 py-2.5",
+      base: "{UInput} text-sm border-brand-100 px-4 py-2.5",
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/configs/default.config.js:64]: there are more Vueless components in this component.

File: src/ui.form-date-picker-range/composables/attrs.composable.js [hunk 15]
```diff
@@ -7,124 +7,136 @@ import defaultConfig from "../configs/default.config";
...
-    base: menu.base,
-    variants: menu.variants,
-    compoundVariants: menu.compoundVariants,
-  });
-
   const menuClasses = computed(() =>
-    cvaMenu({ openDirectionY: openDirectionY.value, openDirectionX: openDirectionX.value }),
+    cva(config.value.menu)({
+      openDirectionY: openDirectionY.value,
+      openDirectionX: openDirectionX.value,
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/composables/attrs.composable.js:19]: same as in UDatePicker

File: src/ui.form-date-picker-range/composables/attrs.composable.js [hunk 16]
```diff
@@ -7,124 +7,136 @@ import defaultConfig from "../configs/default.config";
...
+    });
+
+    attrs[`${key}Attrs`] = getAttrs(key, { classes });
+
+    if (key === "menu") {
+      const menuAttrs = attrs[`${key}Attrs`];
+
+      attrs[`${key}Attrs`] = computed(() => ({
+        ...menuAttrs.value,
+        class: cx([menuClasses.value, menuAttrs.value.class]),
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/composables/attrs.composable.js:45]: this if condition no needed at all

File: src/ui.form-date-picker-range/composables/attrs.composable.js [hunk 17]
```diff
@@ -7,124 +7,136 @@ import defaultConfig from "../configs/default.config";
...
+      }));
     }
 
-    if (calendarAttrs.value.config.i18n || !props.config.i18n) {
-      return;
+    if (key === "input") {
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/composables/attrs.composable.js:49]: inputBlurAttrs no needed

inputActive >>> inputFocus (rename in config)

```js
if (key === "input") {
      const inputAttrs = attrs[`${key}Attrs`];

      attrs[`${key}Attrs`] = computed(() => {
        ...inputAttrs.value,
        class: cx([
          inputAttrs.value.class,
          isShownMenu.value && config.value.inputFocus,
        ]),
      });
    }
```

File: src/ui.form-date-picker-range/composables/attrs.composable.js [hunk 18]
```diff
@@ -7,124 +7,136 @@ import defaultConfig from "../configs/default.config";
...
 
-  const buttonWrapperAttrs = computed(() => ({
-    ...buttonWrapperAttrsRaw.value,
-    class: cx([
-      buttonWrapperAttrsRaw.value.class,
-      isShownMenu.value && config.value.buttonWrapperActive,
-    ]),
-  }));
+    if (key === "periodDate") {
+      attrs[`${key}Attrs`] = (classes = []) => {
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/composables/attrs.composable.js:92]: (classes = []) >>> (classes) - replace in all cases

File: src/ui.form-date-picker-range/composables/attrs.composable.js [hunk 19]
```diff
@@ -7,124 +7,136 @@ import defaultConfig from "../configs/default.config";
...
-    ...buttonAttrsRaw.value,
-    class: cx([buttonAttrsRaw.value.class, isShownMenu.value && config.value.buttonActive]),
-  }));
+    if (key === "calendar") {
+      const calendarAttrs = attrs[`${key}Attrs`];
+
+      // This watcher rewrites default calendar locales with datepicker range locales
+      // Watcher will not rewrite custom calendar locales
+      watchEffect(() => {
+        if (!calendarAttrs.value.config) {
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/composables/attrs.composable.js:103]: same as in date picker

File: src/ui.form-date-picker-range/composables/attrs.composable.js (hunk 16 above)
Comment (KinduD21) [reply src/ui.form-date-picker-range/composables/attrs.composable.js]: I left this if condition in latest commit as it's the only place where these constants with props **"isTop"/"isRight"** are used:
```
const openDirectionY = computed(() => (isTop.value ? POSITION.top : POSITION.bottom));
const openDirectionX = computed(() => (isRight.value ? POSITION.right : POSITION.left));
```

File: src/ui.form-calendar/configs/default.config.js [hunk 20]
```diff
@@ -14,12 +14,12 @@ export default /*tw*/ {
...
-  nextPrevButton: "",
-  nextIcon: "",
+  nextPrevButton: "{UButton}",
+  nextIcon: "{UIcon}",
   nextIconName: "keyboard_arrow_right",
-  prevIcon: "",
+  prevIcon: "{UIcon}",
   prevIconName: "keyboard_arrow_left",
-  dayView: "",
+  dayView: "{DayView}",
```
Comment (itsJohnnyGrid) [inline src/ui.form-calendar/configs/default.config.js:22]: {DayView}  {MonthView} {YearView} internal components no need to be placed here.

File: src/ui.form-date-picker-range/composables/attrs.composable.js [hunk 21]
```diff
@@ -42,17 +35,23 @@ export default function useAttrs(props, { isShownMenu, isTop, isRight }) {
 
       attrs[`${key}Attrs`] = computed(() => ({
         ...menuAttrs.value,
-        class: cx([menuClasses.value, menuAttrs.value.class]),
+        class: cx([
+          cva(config.value.menu)({
+            openDirectionY: openDirectionY.value,
+            openDirectionX: openDirectionX.value,
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/composables/attrs.composable.js:41]: I mean this one: 

```
openDirectionY: isTop.value ? POSITION.top : POSITION.bottom,
openDirectionX: isRight.value ? POSITION.right : POSITION.left,
```

File: src/ui.form-date-picker/composables/attrs.composable.js [hunk 22]
```diff
@@ -50,21 +43,27 @@ export default function useAttrs(props, { isShownCalendar, isTop, isRight }) {
 
       attrs[`${key}Attrs`] = computed(() => ({
         ...calendarAttrs.value,
-        class: cx([calendarWrapperClasses.value, calendarAttrs.value.class]),
+        class: cx([
+          cva(config.value.calendar.wrapper)({
+            openDirectionY,
+            openDirectionX,
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker/composables/attrs.composable.js:49]: fix the same way as I mentioned before

File: src/ui.form-date-picker-range/configs/default.config.js [hunk 23]
```diff
@@ -34,17 +34,17 @@ export default /*tw*/ {
   },
   periodsRow: "mb-1 flex min-w-64 gap-1",
   periodButton: `
-    h-[3.125rem] w-full flex flex-col items-center justify-center
+  {UButton} h-[3.125rem] w-full flex flex-col items-center justify-center
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/configs/default.config.js:37]: the tab is missing

File: src/ui.form-date-picker/configs/default.config.js [hunk 24]
```diff
@@ -1,7 +1,7 @@
 export default /*tw*/ {
   wrapper: "relative",
   input: "",
-  inputActive: {
+  inputFocus: {
     block: {
       base: "ring-4 ring-brand-600/[.15] border-brand-500 hover:border-brand-500",
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker/configs/default.config.js:6]: base nesting no needs

**Review by itsJohnnyGrid — APPROVED** [review]

**Pull request merged**